notifications_python_client/notifications.py

```diff
@@ -50,6 +54,26 @@ def get_all_notifications(self, status=None, template_type=None, reference=None)
             params=data
         )
 
+    def get_all_notifications_iterator(self, status=None, template_type=None, reference=None, older_than=None):
```

Comment:
Since this is literally a wrapper around `get_all_notifications`, the logic for error handling is delegated there. In other words this doesn't introduce any new error handling. `get_all_notifications` does raise on an API error however.